### DISCUSSION TITLE
IgxGridComponent - ability to disable rows in the grid (or: visually differentiate rows based on some kind of state)?

### DISCUSSION THREAD

**Discussion by HendrikHaase** [IgniteUI/igniteui-angular#9862] [Q&A]
Hi,

I want to display data inside a grid, but i need to disable a "row" on certain conditions (not cell-editing, i need to visually make the whole row non-interactive).

After some research it seems like i'm all out of luck on this issue.

The official documentation results in a 404 upon navigating to the IgxGridRow Api Reference ( https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxgridrow.html ).

When logging a row the properties are also not very helpful for my desire to "set a state of a row"..
![image](https://user-images.githubusercontent.com/8005904/125657699-c2663544-6a93-4e08-a12d-ce8ac8f9458e.png)

Can anyone help / throw me a curveball?

regards,
Hendrik

**Comment by zdrawku** [comment]
Hey @HendrikHaase, with the [12.2.0-alpha.2](https://github.com/IgniteUI/igniteui-angular/releases/tag/12.2.0-alpha.2) release we expose a way to style the rows - new 'rowStyles' and 'rowClasses' grid properties which allows to define a custom styling on each grid row.

As for the [igxGridRow API Docs](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxgridrow.html), now you should be able to open it.

If you want to make a row non-editable (not only visually) you can leverage from the [Event Cancelation example](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/editing#event-cancelation) that we have.

- **RowEditEnter** - Neither Row nor Cell will enter edit mode.
- **CellEditEnter** - Prevents entering cell edit. If rowEditable is enabled, row edit will be triggered, although cell edit will remain forbidden.
- **CellEdit** - Allowed Cell/Row edit, hitting Done button or Enter won't commit the value or row transaction. Cell editing and Row editing won't be closed until Cancel button is clicked.
- **RowEdit** - Committing cell is possible, but not the whole row. The row will stay in edit mode and the row transaction will be considered open. Hitting Done does not commit or close the row. Cancel button closes the editing process and the transaction without committing the changes.
